src/EnergyPlus/ThermalComfort.cc

```diff
@@ -3015,22 +3016,33 @@ namespace ThermalComfort {
             const bool epwFileExists = FileSystem::fileExists(state.files.inputWeatherFileName.fileName);
             readStat = 0;
             if (epwFileExists) {
-                auto epwFile = state.files.inputWeatherFileName.open(state, "CalcThermalComfortAdaptiveCEN15251");
-                for (i = 1; i <= 9; ++i) { // Headers
-                    epwFile.readLine();
+                // determine number of days in year
+                int DaysInYear;
+                if (state.dataEnvrn->CurrentYearIsLeapYear) {
+                    DaysInYear = 366;
+                } else {
+                    DaysInYear = 365;
                 }
+
+                // Borrowed from OpenEPlusWeatherFile
+                // state.files.inputWeatherFile.close();
+                // state.files.inputWeatherFile.fileName = state.files.inputWeatherFileName.fileName;
+                // state.files.inputWeatherFile.open();
+                // WeatherManager::SkipEPlusWFHeader(state);
+                WeatherManager::OpenEPlusWeatherFile(state, ErrorsFound, false);
```

Comment:
I am not entirely sure about the relationship between `state.files.inputWeatherFileName` and `state.files.inputWeatherFile`, but in order to use `SkipEPlusWFHeader`, it seems the former needs to be changed to the latter in ASH55 and CEN15251 calculation, and all `epwFile.readLine()` to `state.files.inputWeatherFile.readLine()` accordingly. I am not that certain if this would affect other bits of codes, which I am not familiar with, all tests are passed from my end though.

To use `SkipEPlusWFHeader`, either to directly use `OpenEPlusWeatherFile` with an additional local variable `bool ErrorsFound(false)` (the current version), or to use the four lines from `OpenEPlusWeatherFile` (commented out), which would be preferable?